application/src/main/java/org/togetherjava/tjbot/commands/moderation/ReportCommand.java

```diff
@@ -0,0 +1,196 @@
+
+import java.awt.*;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Predicate;
+import java.util.regex.Pattern;
+
+
+/**
+ * Implements the /report command, which allows users to contact a moderator within the server which
+ * forwards messages to moderators in a dedicated channel given by
+ * {@link Config#getModMailChannelPattern()}.
+ *
+ * The /report command would give us necessary information about the offending user, and specific
+ * message that was offensive.
+ */
+public final class ReportCommand extends BotCommandAdapter implements MessageContextCommand {
+
+    private static final Logger logger = LoggerFactory.getLogger(ReportCommand.class);
+    private static final String COMMAND_NAME = "report";
+    private static final String REPORT_REASON_INPUT_ID = "reportReason";
+    private static final int COOLDOWN_DURATION_VALUE = 3;
+    private static final ChronoUnit COOLDOWN_DURATION_UNIT = ChronoUnit.MINUTES;
+    private static final Color AMBIENT_COLOR = Color.BLACK;
+    private final Cache<Long, Instant> authorToLastReportInvocation = createCooldownCache();
+    private final Predicate<String> modMailChannelNamePredicate;
+    private final String configModMailChannelPattern;
+
+    /**
+     * Creates a new instance.
+     *
+     * @param config to get the channel to forward reports to
+     */
+    public ReportCommand(Config config) {
+        super(Commands.message(COMMAND_NAME), CommandVisibility.GUILD);
+
+        modMailChannelNamePredicate =
+                Pattern.compile(config.getModMailChannelPattern()).asMatchPredicate();
+
+        configModMailChannelPattern = config.getModMailChannelPattern();
+    }
+
+    private Cache<Long, Instant> createCooldownCache() {
+        return Caffeine.newBuilder()
+            .maximumSize(1_000)
+            .expireAfterAccess(COOLDOWN_DURATION_VALUE, TimeUnit.of(COOLDOWN_DURATION_UNIT))
+            .build();
+    }
+
+    @Override
+    public void onMessageContext(MessageContextInteractionEvent event) {
+        long userID = event.getUser().getIdLong();
+
+        if (handleIsOnCooldown(userID, event)) {
+            return;
+        }
+        authorToLastReportInvocation.put(userID, Instant.now());
+
+        String reportedMessage = event.getTarget().getContentRaw();
+        String reportedMessageUrl = event.getTarget().getJumpUrl();
+
+        TextInput body = TextInput
+            .create(REPORT_REASON_INPUT_ID, "Anonymous report to the moderators",
+                    TextInputStyle.PARAGRAPH)
+            .setPlaceholder("What is wrong with the message, why do you want to report it?")
+            .setRequiredRange(3, 200)
+            .build();
+
+        Modal modal =
+                Modal.create(generateComponentId(reportedMessage, reportedMessageUrl), "Report")
```

Comment:
* "Report this to a moderator"
* might make sense to move the component id into a variable right before, just to give it more attention `String componentId = generateComponentId(...);` ... `Modal.create(componentId, ...)`